t/sgunmap-perf.py

```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.7
+#!/usr/bin/env python
```

Comment:
I think we should just go with
```
#!/usr/bin/env python3
```

across the board. Are people really stuck on systems where there's no python3 option at all (OK it's not "easy" to get python3 going on CentOS 7 but it does exist)?

I think these were the commands that @vincentkfu said he was unable to test - do you know if they work with python3?

Reply:
Isn't it a bit early to drop CentOS 6 and CentOS 7 support?

The t/sgunmap-perf.py script has been tested as follows:
```
# modprobe scsi_debug delay=0 dev_size_mb=256
# lsscsi -g
[2:0:0:0]    cd/dvd  QEMU     QEMU DVD-ROM     2.5+  /dev/sr0   /dev/sg0 
[8:0:0:0]    disk    Linux    scsi_debug       0188  /dev/sda   /dev/sg1 
# t/sgunmap-test.py /dev/sg1 /dev/sda ./fio
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg1', '--iodepth=32', '--iodepth_batch=2']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg1', '--iodepth=32', '--iodepth_batch=4']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg1', '--iodepth=32', '--iodepth_batch=8']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg1', '--iodepth=64', '--iodepth_batch=4']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg1', '--iodepth=64', '--iodepth_batch=8']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg1', '--iodepth=64', '--iodepth_batch=16']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg1', '--iodepth=128', '--iodepth_batch=8']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg1', '--iodepth=128', '--iodepth_batch=16']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg1', '--iodepth=128', '--iodepth_batch=32']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randread', '--filename=/dev/sg1', '--iodepth=1', '--iodepth_batch=1']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randwrite', '--filename=/dev/sg1', '--iodepth=1', '--iodepth_batch=1']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg1', '--iodepth=1', '--iodepth_batch=1']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randread', '--filename=/dev/sda', '--iodepth=1', '--iodepth_batch=1']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randwrite', '--filename=/dev/sda', '--iodepth=1', '--iodepth_batch=1']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sda', '--iodepth=1', '--iodepth_batch=1']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randread', '--filename=/dev/sg1', '--iodepth=16', '--iodepth_batch=2']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randwrite', '--filename=/dev/sg1', '--iodepth=16', '--iodepth_batch=2']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg1', '--iodepth=16', '--iodepth_batch=2']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randread', '--filename=/dev/sda', '--iodepth=16', '--iodepth_batch=2']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randwrite', '--filename=/dev/sda', '--iodepth=16', '--iodepth_batch=2']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sda', '--iodepth=16', '--iodepth_batch=2']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randread', '--filename=/dev/sg1', '--iodepth=16', '--iodepth_batch=16']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randwrite', '--filename=/dev/sg1', '--iodepth=16', '--iodepth_batch=16']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg1', '--iodepth=16', '--iodepth_batch=16']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randread', '--filename=/dev/sda', '--iodepth=16', '--iodepth_batch=16']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randwrite', '--filename=/dev/sda', '--iodepth=16', '--iodepth_batch=16']
**********passed*********
['--name=test', '--time_based', '--runtime=30s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sda', '--iodepth=16', '--iodepth_batch=16']
**********passed*********
```

Reply:
Oh I didn't know that the scsi_debug module existed. I tried it with t/sgunmap-test.py and saw an issue with the test script which is fixed by [this patch](https://github.com/vincentkfu/fio/commit/aef939adc1692b3b9f1c6fba58b91d5bf5e3ed2d).

```
# modprobe scsi_debug delay=0 dev_size_mb=256
# lsscsi -g
[3:0:0:0]    cd/dvd  ASUS     BW-16D1HT        3.00  /dev/sr0   /dev/sg0 
[5:0:0:0]    cd/dvd  ASUS     BC-12B1ST        1.01  /dev/sr1   /dev/sg5 
[8:0:0:0]    cd/dvd  ASUS     BC-08B1LT        1.03  /dev/sr2   /dev/sg6 
[12:0:0:0]   disk    Generic  MassStorageClass 1210  /dev/sda   /dev/sg1 
[12:0:0:1]   disk    Generic  MassStorageClass 1210  /dev/sdb   /dev/sg2 
[12:0:0:2]   disk    Generic  MassStorageClass 1210  /dev/sdc   /dev/sg3 
[12:0:0:3]   disk    Generic  MassStorageClass 1210  /dev/sdd   /dev/sg4 
[13:0:0:0]   disk    Linux    scsi_debug       0188  /dev/sde   /dev/sg7 
# python2.7 t/sgunmap-test.py /dev/sg7 /dev/sde ./fio
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg7', '--iodepth=32', '--iodepth_batch=2']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg7', '--iodepth=32', '--iodepth_batch=4']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg7', '--iodepth=32', '--iodepth_batch=8']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg7', '--iodepth=64', '--iodepth_batch=4']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg7', '--iodepth=64', '--iodepth_batch=8']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg7', '--iodepth=64', '--iodepth_batch=16']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg7', '--iodepth=128', '--iodepth_batch=8']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg7', '--iodepth=128', '--iodepth_batch=16']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg7', '--iodepth=128', '--iodepth_batch=32']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randread', '--filename=/dev/sg7', '--iodepth=1', '--iodepth_batch=1']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randwrite', '--filename=/dev/sg7', '--iodepth=1', '--iodepth_batch=1']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg7', '--iodepth=1', '--iodepth_batch=1']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randread', '--filename=/dev/sde', '--iodepth=1', '--iodepth_batch=1']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randwrite', '--filename=/dev/sde', '--iodepth=1', '--iodepth_batch=1']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sde', '--iodepth=1', '--iodepth_batch=1']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randread', '--filename=/dev/sg7', '--iodepth=16', '--iodepth_batch=2']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randwrite', '--filename=/dev/sg7', '--iodepth=16', '--iodepth_batch=2']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg7', '--iodepth=16', '--iodepth_batch=2']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randread', '--filename=/dev/sde', '--iodepth=16', '--iodepth_batch=2']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randwrite', '--filename=/dev/sde', '--iodepth=16', '--iodepth_batch=2']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sde', '--iodepth=16', '--iodepth_batch=2']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randread', '--filename=/dev/sg7', '--iodepth=16', '--iodepth_batch=16']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randwrite', '--filename=/dev/sg7', '--iodepth=16', '--iodepth_batch=16']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sg7', '--iodepth=16', '--iodepth_batch=16']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randread', '--filename=/dev/sde', '--iodepth=16', '--iodepth_batch=16']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randwrite', '--filename=/dev/sde', '--iodepth=16', '--iodepth_batch=16']
**********passed*********
['--max-jobs=4', '--name=test', '--time_based', '--runtime=3s', '--output-format=json', '--ioengine=sg', '--rw=randtrim', '--filename=/dev/sde', '--iodepth=16', '--iodepth_batch=16']
**********passed*********
```